app/queries/servers/thread.ts

```diff
@@ -0,0 +1,125 @@
+
+export function processIsCRTEnabled(preferences: PreferenceModel[], config?: ClientConfig): boolean {
+    let preferenceDefault = Preferences.COLLAPSED_REPLY_THREADS_OFF;
+    const configValue = config?.CollapsedThreads;
+    if (configValue === 'default_on') {
+        preferenceDefault = Preferences.COLLAPSED_REPLY_THREADS_ON;
+    }
+    const preference = getPreferenceValue(preferences, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.COLLAPSED_REPLY_THREADS, preferenceDefault);
+
+    const isAllowed = config?.FeatureFlagCollapsedThreads === 'true' && config?.CollapsedThreads !== 'disabled';
+
+    return isAllowed && (preference === Preferences.COLLAPSED_REPLY_THREADS_ON || config?.CollapsedThreads === 'always_on');
+}
+
+export const getIsCRTEnabled = async (database: Database): Promise<boolean> => {
+    const {value: config} = await database.get<SystemModel>(SYSTEM).find(SYSTEM_IDENTIFIERS.CONFIG);
+    const preferences = await database.get<PreferenceModel>(PREFERENCE).query(Q.where('category', Preferences.CATEGORY_DISPLAY_SETTINGS)).fetch();
+    return processIsCRTEnabled(preferences, config);
+};
+
+export const getThreadById = async (database: Database, threadId: string) => {
+    try {
+        const thread = await database.get<ThreadModel>(THREAD).find(threadId);
+        return thread;
+    } catch {
+        return undefined;
+    }
+};
+
+export const observeIsCRTEnabled = (database: Database) => {
+    const config = database.get<SystemModel>(SYSTEM).findAndObserve(SYSTEM_IDENTIFIERS.CONFIG);
+    const preferences = database.get<PreferenceModel>(PREFERENCE).query(Q.where('category', Preferences.CATEGORY_DISPLAY_SETTINGS)).observe();
+    return combineLatest([config, preferences]).pipe(
+        map(
+            ([{value: cfg}, prefs]) => processIsCRTEnabled(prefs, cfg),
+        ),
+    );
+};
+
+export const queryThreadById = (database: Database, threadId: string) => {
+    return database.get<ThreadModel>(THREAD).query(
+        Q.where('id', threadId),
+    ).observe().pipe(
+        switchMap((threads) => threads[0]?.observe() || of$(undefined)),
+    );
+};
+
+type QueryThreadsInTeamArgs = {
+    database: Database;
+    hasReplies?: boolean;
+    isFollowing?: boolean;
+    onlyUnreads?: boolean;
+    sort?: boolean;
+    teamId?: string;
+};
+
+export const queryThreadsInTeam = ({database, hasReplies = true, isFollowing = true, onlyUnreads, sort, teamId}: QueryThreadsInTeamArgs): Query<ThreadModel> => {
+    const query: Q.Clause[] = [];
+
+    if (isFollowing) {
+        query.push(Q.where('is_following', true));
+    }
+
+    // Posts can be followed even without replies, they shouldn't be displayed in global threads
+    if (hasReplies) {
+        query.push(Q.where('reply_count', Q.gt(0)));
+    }
+
+    // If teamId is specified, only get threads in that team and DM
+    if (teamId) {
+        query.push(
+            Q.experimentalNestedJoin(POST, CHANNEL),
```

Comment:
In the examples, this clause is usually at the beginning of the clauses. Not sure if the order is important in any way.

Reply:
did you ended up looking at the query plan, are we sure this is actually working?

I mean if you create the query without the array as one single query, does TS yell at you?